pandas/core/indexing.py

```diff
@@ -2007,6 +2007,11 @@ def _convert_key(self, key, is_setter: bool = False):
         Require they keys to be the same type as the index. (so we don't
         fallback)
         """
+        # For series, unpacking key needs to result in the label.
+        # This is already the case for non-multiindex
+        if isinstance(self.obj, ABCSeries) and len(key) > 1:
```

Comment:
I'm still confused.  Dont we want to do the tuple-wrapping only if we dont have MultiIndex?

Reply:
Ah, I now understand what you were asking before. Currently on master, if one attempts to access a series without a multiindex and len(keys) > 1, then the ValueError

    Not enough indexers for scalar access (setting)!

would be raised. Note that there are actually too *many* indexers for scalar access in this case. On this branch, because of the tuple-wrapping, a KeyError would instead be raised in the same way as searching for any label that is not in the series. It seems to me this latter behavior is desired. Happy to change it if that's not the case.

Reply:
I should have also mentioned the .get access in my previous response. Here a TypeError would be raised 

     _get_value() got multiple values for argument 'takeable'

whereas again this PR would change that to a KeyError.

Reply:
what im suggesting is that the check here should include `isinstance(self.obj.index, MultiIndex)`

Reply:
Completely understood. In my previous response, I was trying to detail how I felt that not adding your suggested check would improve the error messages. I was of the impression that a series without a multiindex and len(keys) > 1 would necessarily result in an error, but I now see that I was wrong:

    series = pd.Series([1, 2], index=[(1, 2), (3, 4)])
    series.at[1, 2]

The above code will fail on master but work in this PR. With your suggested check, it will also fail on this PR. I've added tests for this case, but it is unclear to me if pandas supports having tuples in the index. For example, in this case series.loc[(1, 2)] fails (both on master and in this PR).

Reply:
@jbrockmendel you ok here?